recipes/cmake/3.x.x/conanfile.py

```diff
@@ -55,85 +56,88 @@ def validate(self):
-                                  "project(CMake)",
-                                  "project(CMake)\ninclude(\"{}/conanbuildinfo.cmake\")\nconan_basic_setup(NO_OUTPUT_DIRS)".format(
-                                      self.install_folder.replace("\\", "/")))
-            if self.settings.os == "Linux":
-                tools.replace_in_file(os.path.join(self._source_subfolder, "Utilities", "cmcurl", "CMakeLists.txt"),
-                                      "list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES})",
-                                      "list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES} ${CMAKE_DL_LIBS} pthread)")
-
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.build()
 
     def package(self):
-        self.copy("Copyright.txt", dst="licenses", src=self._source_subfolder)
+        copy(self, "Copyright.txt", self.source_folder, os.path.join(self.package_folder, "licenses"), keep_path=False)
         if self.options.bootstrap:
-            with tools.chdir(self._source_subfolder):
-                autotools = AutoToolsBuildEnvironment(self)
+            with chdir(self, self.source_folder):
+                autotools = Autotools(self)
                 autotools.install()
         else:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.install()
         rmdir(self, os.path.join(self.package_folder, "doc"))
 
     def package_id(self):
         del self.info.settings.compiler
+        del self.info.options.bootstrap
 
     def package_info(self):
-        module_version = "{}.{}".format(Version(self.version).major, Version(self.version).minor)
-
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bindir))
-        self.env_info.PATH.append(bindir)
-
         self.buildenv_info.prepend_path("CMAKE_ROOT", self.package_folder)
-        self.env_info.CMAKE_ROOT = self.package_folder
+        self.runenv_info.define_path("CMAKE_ROOT", self.package_folder)
+        module_version = "{}.{}".format(Version(self.version).major, Version(self.version).minor)
         mod_path = os.path.join(self.package_folder, "share", f"cmake-{module_version}", "Modules")
         self.buildenv_info.prepend_path("CMAKE_MODULE_PATH", mod_path)
-        self.env_info.CMAKE_MODULE_PATH = mod_path
+        self.runenv_info.define_path("CMAKE_MODULE_PATH", mod_path)
         if not os.path.exists(mod_path):
             raise ConanException("Module path not found: %s" % mod_path)
 
         self.cpp_info.includedirs = []
+
+        # TODO remove once conan v2 is the only support and recipes have been migrated
+        bindir = os.path.join(self.package_folder, "bin")
+        self.output.info("Appending PATH environment variable: {}".format(bindir))
+        self.runenv_info.append_path("PATH", bindir)
```

Comment:
the bin folder should automatically get aded to the PATH when using `VirtualRunEnv` so to my my knowledge this should not be required 🤞 

```suggestion
```